Release Notes/510.md

```diff
@@ -11,6 +11,11 @@
 
 ## API Behavior Changes
 
+- Leaf Node Protocols
+  - Description: Syntax nodes that do not act as base nodes for other syntax types now conform to a new protocols that deprecates inherited casting methods (e.g. `AccessorBlockSyntax` to `LeafSyntaxNodeProtocol` or `AccessorDeclSyntax` to `LeafDeclSyntaxNodeProtocol`). This change aims to prevent unsafe type-casting by issuing deprecation warnings for methods that will result in failed casts.
```

Comment:
I would try to avoid the introduction of the protocols since it’s just a technical detail.

```suggestion
  - Description: Syntax nodes that do not act as base nodes for other syntax types have the casting methods marked as deprecated. This prevents unsafe type-casting by issuing deprecation warnings for methods that will always result in failed casts.
```

Reply:
Thanks, done